### PR TITLE
Added translation of property name for nested content property editors.

### PR DESCRIPTION
**Pull request by cheeseytoastie**
Mandatory messages are not changed and the text "Cannot be empty" remains untranslated which seems wrong? But this fixes the issue reported by 8081

### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/8081

### Description
The issue described in 8081 needed the LocalizedTextService passed in through to the mapper so we can map the "label" of the property name in the same way it works for the vanilla property editor.  

I didn't attempt to translate ".... cannot be empty" though it seems like this would be a good addition to the Umbraco dictionary. 

To test:
1) Create two users, EN / DK
2) Create a Nested Content Doc type with a mandatory field name as per the instructions in the issue (namely named #p_NameTest and create corresponding dictionary item "p_NameTest" with EN and DK translations. 
3) Attempt to save and publish a new NC item without filling in the mandatory field. 
4) Check that the warning has the translated field name instead of #p_NameTest rendered.

### CODE REVIEW

**Comment by cheeseytoastie** [comment]
Oops I've broke a test - will fix tomorrow

**Comment by joepvtl** [comment]
Hi,

Whilst you are at it, shouldn't you also add the translations to these two lines aswell?
https://github.com/umbraco/Umbraco-CMS/blob/b2a9e17c90ab5c9d2bfa7c7b6d2e738a152073e0/src/Umbraco.Web/PropertyEditors/NestedContentPropertyEditor.cs#L287
https://github.com/umbraco/Umbraco-CMS/blob/b2a9e17c90ab5c9d2bfa7c7b6d2e738a152073e0/src/Umbraco.Web/PropertyEditors/NestedContentPropertyEditor.cs#L298
-Joep

**Comment by cheeseytoastie** [comment]
> Hi,
> 
> Whilst you are at it, shouldn't you also add the translations to these two lines aswell?
> https://github.com/umbraco/Umbraco-CMS/blob/b2a9e17c90ab5c9d2bfa7c7b6d2e738a152073e0/src/Umbraco.Web/PropertyEditors/NestedContentPropertyEditor.cs#L287
> 
> 
> https://github.com/umbraco/Umbraco-CMS/blob/b2a9e17c90ab5c9d2bfa7c7b6d2e738a152073e0/src/Umbraco.Web/PropertyEditors/NestedContentPropertyEditor.cs#L298
> 
> 
> -Joep

Good spot - added those

File: src/Umbraco.Web/PropertyEditors/NestedContentPropertyEditor.cs [hunk 1]
```diff
@@ -31,14 +31,16 @@ public class NestedContentPropertyEditor : DataEditor
         private readonly Lazy<PropertyEditorCollection> _propertyEditors;
         private readonly IDataTypeService _dataTypeService;
         private readonly IContentTypeService _contentTypeService;
+        private readonly ILocalizedTextService _textService;
         internal const string ContentTypeAliasPropertyKey = "ncContentTypeAlias";
 
-        public NestedContentPropertyEditor(ILogger logger, Lazy<PropertyEditorCollection> propertyEditors, IDataTypeService dataTypeService, IContentTypeService contentTypeService)
+        public NestedContentPropertyEditor(ILogger logger, Lazy<PropertyEditorCollection> propertyEditors, IDataTypeService dataTypeService, IContentTypeService contentTypeService, ILocalizedTextService textService)
```
Comment (ronaldbarendse) [inline src/Umbraco.Web/PropertyEditors/NestedContentPropertyEditor.cs:37]: This should be added as a new constructor and the old constructor should be obsoleted (and get the `ILocalizedTextService` instance from `Current.Services.TextService`):
```suggestion
        [Obsolete("Use constructor with all dependencies")]
        public NestedContentPropertyEditor(ILogger logger, Lazy<PropertyEditorCollection> propertyEditors, IDataTypeService dataTypeService, IContentTypeService contentTypeService)
            : this(logger, propertyEditors, dataTypeService, contentTypeService, Current.Services.TextService)
        { }
        
        public NestedContentPropertyEditor(ILogger logger, Lazy<PropertyEditorCollection> propertyEditors, IDataTypeService dataTypeService, IContentTypeService contentTypeService, ILocalizedTextService textService)
```

**Pull request closed without merging**